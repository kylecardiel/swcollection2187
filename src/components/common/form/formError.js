import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import React from 'react';
import { red } from '@material-ui/core/colors';

const useStyles = makeStyles(theme => ({
    formError: {
        color: red[700],
        backgroundColor: red[100],
        borderRadius: 5,
        textAlign: 'center',
        borderStyle: 'groove',
        borderWidth: '1px',
        borderColor: red[700],
        fontStyle: 'italic',
    },
}));

export const FormError = props => {

    const classes = useStyles();

    return (
        <Typography data-testid='FormErrorId' component='p' className={classes.formError}>
            {props.errorMessage}
        </Typography>
    );
}
