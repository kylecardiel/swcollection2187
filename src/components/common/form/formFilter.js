import React from 'react';
import FormControl from '@material-ui/core/FormControl';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import InputLabel from '@material-ui/core/InputLabel';
import { makeStyles } from '@material-ui/core/styles';

export const FormFilter = ({ menuList, onChange, label, inputLabel, labelWidth, value }) => {

    const classes = useStyles();
    const menuItemsList = list => {
        return list.map(item =>
            <MenuItem key={item} value={item}>{item}</MenuItem>
        )
    };

    return (
        <FormControl variant='outlined' className={classes.formControl}>
            <InputLabel ref={inputLabel} id={`${label}-id`}>{label}</InputLabel>
            <Select
                labelId={`${label}-id`}
                id={label}
                onChange={onChange}
                labelWidth={labelWidth}
                defaultValue={''}
                label={label}
                value={value}
            >
                <MenuItem key={'none'} value={null}><em>{'none'}</em></MenuItem>
                {menuItemsList(menuList)}
            </Select>
        </FormControl>
    )
}

const useStyles = makeStyles((theme) => ({
    formControl: {
        margin: theme.spacing(1),
        minWidth: 225,
    },
}));