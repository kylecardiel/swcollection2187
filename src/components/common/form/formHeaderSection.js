import { makeStyles } from '@material-ui/core/styles';
import React from 'react';
import { Color } from 'shared/styles/color';
import Grid from '@material-ui/core/Grid';
import { HeaderText } from 'components/common/text/headerText';
import CancelIcon from '@material-ui/icons/Cancel';

export const FormHeaderSection = props => {
    const classes = useStyles();
    const { text, textColor } = props;

    return (
        <Grid data-testid='FormHeaderId' container spacing={1} className={classes.headerContainer}>
            <Grid item xs={12} >
                <HeaderText text={text} textColor={textColor}/>
            </Grid>
            {/* <Grid item xs={1} className={classes.cancelGrid}>
                <CancelIcon fontSize='large' style={{ color:'white' }} />
            </Grid> */}
        </Grid>
    );
};

const useStyles = makeStyles(theme => ({
    headerContainer: {
        paddingTop: theme.spacing(3),
        paddingBottom: theme.spacing(3),
        background: Color.primary('eliteBlackGradient'),
    },
    cancelGrid: {
        display: 'flex,',
        justifyContent: 'flex-end',
        alignItems: 'right',
    },
    cancelIconColor: {
        color: 'white',
    },
}));
