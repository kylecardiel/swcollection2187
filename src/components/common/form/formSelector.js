import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import { makeStyles } from '@material-ui/core/styles';
import { forEach } from 'lodash';
import React from 'react';
import { Color } from 'shared/styles/color';

const useStyles = makeStyles(theme => ({
    '@global': {
        body: {
            backgroundColor: theme.palette.common.white
        }
    },
    form: {
        width: '100%',
        marginTop: theme.spacing(1),
        minWidth: 120,
        backgroundColor: props => props.backgroundColor,
    },
}));

export const FormSelector = props => {

    const inputLabel = React.useRef(null);
    const [labelWidth, setLabelWidth] = React.useState(0);
    React.useEffect(() => {
      setLabelWidth(inputLabel.current.offsetWidth);
    }, []);

    const classProps = { backgroundColor: props.disabled ? Color.lightGrey() : Color.white() };
    const classes = useStyles(classProps);

    let menuItems = [];
    const menuItemsList = () => {
        forEach(props.menuItems, menuItem => {
            menuItems.push(
                <MenuItem key={menuItem.value} value={menuItem.value}>
                    {menuItem.label}
                </MenuItem>
            );
        });
        return menuItems;
    };

    return (
        <FormControl variant="outlined" className={classes.form}>
            <InputLabel ref={inputLabel} id="demo-simple-select-outlined-label">
                {props.inputLabel}
            </InputLabel>
            <Select
                labelId="demo-simple-select-outlined-label"
                id="demo-simple-select-outlined"
                labelWidth={labelWidth}
                // onChange={props.handleChange}
                disabled={props.disabled}
                inputProps={props.inputProps}
            >
                <MenuItem value=''>
                    <em>None</em>
                </MenuItem>
                 {menuItemsList()}
            </Select>
        </FormControl>
    );
}
