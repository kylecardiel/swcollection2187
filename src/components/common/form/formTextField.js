import React from 'react';
import TextField from '@material-ui/core/TextField';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
    main: {
        backgroundColor: props => props.backgroundColor,
    },
}));

export const FormTextField = props => {

    const { backgroundColor } = props;
    const classProps = { backgroundColor: backgroundColor };
    const classes = useStyles(classProps);

    return (
        <TextField
            className={classes.main}
            label={props.label}
            name={props.label.toLowerCase()}
            id={props.name}
            type={props.type}
            inputRef={props.register}
            autoFocus={props.autoFocus}
            variant='outlined'
            required
            fullWidth
            InputLabelProps={{
                shrink: true,
            }}
        />

    );
}
