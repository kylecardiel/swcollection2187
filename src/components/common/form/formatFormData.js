export const formatFormData = snapshot => {
    return {
        assortment: createFormDataObj(snapshot.assortment),
        characters: createFormDataObj(snapshot.characters),
        collectionType: createFormDataObj(snapshot.collectionType),
        exclusiveRetailer: createFormDataObj(snapshot.exclusiveRetailer),
        groups: createFormDataObj(snapshot.groups),
        packageType: createFormDataObj(snapshot.packageType),
        series: createFormDataObj(snapshot.series),
        sourceMaterial: createFormDataObj(snapshot.sourceMaterial),
        sourceType: createFormDataObj(snapshot.sourceType),
        version: createFormDataObj(snapshot.version),

        videoGameConsole: createFormDataObj(snapshot.videoGameConsole),
        videoGameFormat: createFormDataObj(snapshot.videoGameFormat),
        videoGameSeries: createFormDataObj(snapshot.videoGameSeries),
        videoGameType: createFormDataObj(snapshot.videoGameType),
    };
};

const createFormDataObj = retreivedData => {
    let data = {};
    for (let id in retreivedData) {
        data.id = id;
        data.values = retreivedData[id];
    }
    return data;
};

export const convertArrayObjectToArrayOfObjectProperty = (data, attribute) => {
    return data.values.map(obj => obj[attribute]);
};