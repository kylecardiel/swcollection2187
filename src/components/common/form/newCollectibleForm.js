import {
    Button,
    FormControl,
    InputLabel,
    makeStyles,
    TextField,
    MenuItem,
    Select,
    Typography,
    Input,
} from '@material-ui/core';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import React, { useEffect, useRef, useState } from 'react';
import { Controller, useForm } from 'react-hook-form';
import { Color } from 'shared/styles/color';
import { CatalogApi } from 'shared/api/orchestrator';
import { FormHeaderSection } from 'components/common/form/formHeaderSection';
import {
    ALL_ASSORTMENT,
    PRODUCT_LINES,
    PRODUCT_TYPE,
    GROUP_NAMES,
    VERSIONS,
} from 'shared/constants/domainConstantSelectors';
import { FB_DB_CONSTANTS } from 'shared/constants/databaseRefConstants';
import { storage } from 'backend/Firebase';
import { FB_STORAGE_CONSTANTS } from 'shared/constants/storageRefConstants';

const { CATALOG, ACTION_FIGURES } = FB_STORAGE_CONSTANTS;

export const NewCollectibleForm = ({ catalog, closeModal, formData }) => {
    const classes = useStyles();

    const { register, handleSubmit, control } = useForm();

    const [groups, setGroups] = useState([]);
    const [looseFigureImageFile, setLooseFigureImageFile] = useState(null);
    const [looseBlackFigureImageFile, setBlackLooseFigureImageFile] = useState(null);
    const [newFigureImageFile, setNewFigureImageFile] = useState(null);

    const [looseFigureImageDownloadURL, setLooseFigureImageDownloadURL] = useState(null);

    // const [looseFigureImageURL, setLooseFigureImageURL] = useState(null);
    // const [newFigureImageURL, setNewFigureImageURL] = useState(null);

    const handleChange = (event) => {
        setGroups(event.target.value);
    };

    const handleLooseImageChange = e => {
        if (e.target.files[0]) {
            setLooseFigureImageFile(e.target.files[0]);
        }
    };

    const handleBlacLoosekImageChange = e => {
        if (e.target.files[0]) {
            setBlackLooseFigureImageFile(e.target.files[0]);
        }
    };

    const handleNewImageChange = e => {
        if (e.target.files[0]) {
            setNewFigureImageFile(e.target.files[0]);
        }
    };

    const inputLabel = useRef(null);
    const [labelWidth, setLabelWidth] = useState(0);
    useEffect(() => {
        setLabelWidth(inputLabel.current.offsetWidth);
    }, []);


    // const uploadImage = async image => {
    
    //     return storage.ref(`${CATALOG}${ACTION_FIGURES.BLACK_SERIES}${image.name}`).put(image)
    //       .then(snap => {
    //         return snap.ref.getDownloadURL();
    //       })
    //       .then(downloadURL => {
    //         return downloadURL;
    //       })
    //       .catch(error => {
    //         console.log(`An error occurred while uploading the file.\n\n${error}`);
    //       });
    //   }


    const upload = async image => {
        return new Promise((resolve, reject) => {
            // Most of this was copied from firebase example with inputs adjusted for my usecase
            const uploadTask = storage.ref(`${CATALOG}${ACTION_FIGURES.BLACK_SERIES}${image.name}`).put(image);
            // const downloadURL = await uploadTask.ref.getDownloadURL();
            // return downloadURL;
            return uploadTask.on('state_changed',
                snapshot => {
                    var progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
                    console.log('Upload is ' + progress + '% done');
                },
                function error(err) {
                    console.log('error', err);
                    reject();
                },
                function complete() {
                    uploadTask.snapshot.ref.getDownloadURL().then( downloadURL => {
                        // return downloadURL;
                        resolve(downloadURL);
                        // setLooseFigureImageDownloadURL(downloadURL);
                    })
                }
            )
        })
    }


    const onSubmit = async collectible => {
        // const promises = [];

        // const uploadTaskLooseImage = storage.ref(`${CATALOG}${ACTION_FIGURES.BLACK_SERIES}${looseFigureImageFile.name}`).put(looseFigureImageFile);
        // promises.push(uploadTaskLooseImage);

        // await uploadTaskLooseImage.on(
        //     'state_changed',
        //     function (snapshot) {
        //         var progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
        //         console.log('Upload is ' + progress + '% done');
        //     }, function (error) {
        //         console.log(error)
        //     }, async function () {
        //         uploadTaskLooseImage.snapshot.ref.getDownloadURL().then(function (downloadURL) {
        //             console.log('***************************');
        //             console.log(downloadURL)
        //             setLooseFigureImageURL(downloadURL);
        //             console.log('***************************');
        //         });
        //     }
        // );

        // const uploadTaskNewImage = storage.ref(`${CATALOG}${ACTION_FIGURES.BLACK_SERIES}${newFigureImageFile.name}`).put(newFigureImageFile);
        // promises.push(uploadTaskNewImage);

        // uploadTaskNewImage.on(
        //     'state_changed',
        //     function (snapshot) {
        //         var progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
        //         console.log('Upload is ' + progress + '% done');
        //     }, function (error) {
        //         console.log(error)
        //     }, async function () {
        //         uploadTaskNewImage.snapshot.ref.getDownloadURL().then(function (downloadURL) {
        //             console.log('***************************');
        //             console.log(downloadURL)
        //             setNewFigureImageURL(downloadURL);
        //             console.log('***************************');
        //         });
        //     }
        // );

        // Promise.all(promises).then(tasks => {


            collectible.looseImageUrl = await upload(looseFigureImageFile);
            console.log('all image uploads complete');
            collectible.groups = groups;
    
            Object.keys(collectible).forEach(
                key => collectible[key] === undefined && delete collectible[key]
            );
    
            CatalogApi.create(FB_DB_CONSTANTS.ACTION_FIGURES.BLACK_SERIES, collectible)
            closeModal();


        // collectible.looseImageUrl = looseFigureImageDownloadURL;
        // console.log(collectible.looseImageUrl);
        // collectible.newImageUrl = await upload(newFigureImageFile);
        // console.log(collectible.newImageUrl);
        // });
    };

    const menuItemNone = <MenuItem key={'none'} value={null}><em>{'none'}</em></MenuItem>;

    const generatorInputText = text => {
        return <Grid item xs={12} md={4} className={classes.inputBoxInColumn}>
            <Typography variant='subtitle1' className={classes.text}>
                {text}
            </Typography>
        </Grid>;
    };

    const generateSelector = (text, selectorName, selectorValues) => {
        return <>
            {generatorInputText(text)}
            <Grid item xs={12} md={8} className={classes.inputBoxInColumn}>
                <FormControl variant='outlined' className={classes.form}>
                    <InputLabel ref={inputLabel} id={`${selectorName}-label`}>{selectorName}</InputLabel>
                    <Controller
                        name={selectorName}
                        control={control}
                        defaultValue={''}
                        as={
                            <Select
                                label={selectorName}
                                labelId={selectorName}
                                labelWidth={labelWidth}
                                inputProps={{ name: selectorName }}
                            >
                                {menuItemNone}
                                {selectorValues.map(e => <MenuItem key={e} value={e}>{e}</MenuItem>)}
                            </Select>
                        }
                    />
                </FormControl>
            </Grid>
        </>;
    };

    const groupSelect = () => {
        return <>
            {generatorInputText('Groups')}
            <Grid item xs={12} md={8} className={classes.inputBoxInColumn}>
                <FormControl className={classes.formControl}>
                    <InputLabel id="Groups-label">{'Groups'}</InputLabel>
                    <Select
                        labelId="Groups-label"
                        id="groups"
                        multiple
                        value={groups}
                        onChange={handleChange}
                        input={<Input />}
                    >
                        {GROUP_NAMES.map(e => <MenuItem key={e} value={e}>{e}</MenuItem>)}
                    </Select>
                </FormControl>
            </Grid>
        </>;
    };

    const generatorInput = (text, inputName, number) => {
        return <>
            {generatorInputText(text)}
            <Grid item xs={12} md={8} className={classes.inputBoxInColumn}>
                <TextField
                    variant='outlined'
                    className={classes.form}
                    fullWidth
                    id={inputName}
                    name={inputName}
                    label={inputName}
                    inputRef={register()}
                    type={number ? 'number' : 'string'}
                />
            </Grid>
        </>;
    };

    const generatorImageInput = (text, handleChange) => {
        return <>
            {generatorInputText(text)}
            <Grid item xs={12} md={8} className={classes.inputBoxInColumn}>
                <input type='file' onChange={handleChange} />
            </Grid>
        </>;
    };

    const { sourceMaterial } = formData;

    const collectionTypeInput = generateSelector('Collection Type', 'collectionType', PRODUCT_TYPE);
    const seriesTypeInput = generateSelector('Series', 'series', PRODUCT_LINES);
    const assortmentInput = generateSelector('Assortment', 'assortment', ALL_ASSORTMENT);
    const versionTypeInput = generateSelector('Versions', 'version', VERSIONS);
    const sourceMaterialTypeInput = generateSelector('Source Material', 'sourceMaterial', sourceMaterial.values);
    const nameInput = generatorInput('Name', 'name');
    const additionalNameDetailsInput = generatorInput('Additional Name Details', 'additionalNameDetails');
    const seriesNumberInput = generatorInput('Wave', 'wave', true);
    const waveInput = generatorInput('Series Number', 'seriesNumber', true);
    const newInBoxQtyInput = generatorInput('NIB Qty', 'newInBoxQty', true);
    const looseCompleteQtyInput = generatorInput('Loose Complete Qty', 'looseCompleteQty', true);
    const looseIncompleteQtyInput = generatorInput('Loose incmplete Qty', 'looseIncompleteQty', true);
    // const newInBoxImageURLInput = generatorInput('NIB Image URL', 'newImageUrl');
    // const looseImageURLInput = generatorInput('Loose Image URL', 'looseImageUrl');
    const groupSelectInput = groupSelect();
    const purchasePriceInput = generatorInput('Purchase Price', 'purchasePrice');

    const looseImageInput = generatorImageInput('Loose Image', handleLooseImageChange);
    const looseBlackImageInput = generatorImageInput('Loose Black Image', handleBlacLoosekImageChange);
    const newImageInput = generatorImageInput('NIB Image', handleNewImageChange);



    return (
        <React.Fragment>
            <FormHeaderSection text={'Feed the Collection'} textColor={'white'} />
            <Container component='main' maxWidth='xl' className={classes.conatiner}>
                <form onSubmit={handleSubmit(onSubmit)}>
                    <Grid container spacing={1} className={classes.submitButton}>
                        {nameInput}
                        {additionalNameDetailsInput}
                        {collectionTypeInput}
                        {seriesTypeInput}
                        {assortmentInput}
                        {waveInput}
                        {seriesNumberInput}
                        {versionTypeInput}
                        {sourceMaterialTypeInput}
                        {!catalog && newInBoxQtyInput}
                        {!catalog && looseCompleteQtyInput}
                        {!catalog && looseIncompleteQtyInput}
                        {/* {newInBoxImageURLInput} */}
                        {/* {looseImageURLInput} */}
                        {looseImageInput}
                        {looseBlackImageInput}
                        {newImageInput}
                        {groupSelectInput}
                        {!catalog && purchasePriceInput}
                        <Grid item xs={12} className={classes.submitButtonrow}>
                            <Button
                                type='submit'
                                fullWidth
                                variant='contained'
                                className={classes.submit}
                            >
                                {'Submit'}
                            </Button>
                        </Grid>
                    </Grid>
                </form>
            </Container>
        </React.Fragment>
    );
};

const useStyles = makeStyles(theme => ({
    '@global': {
        body: {
            backgroundColor: theme.palette.common.white
        },
    },
    conatiner: {
        paddingTop: theme.spacing(2),
    },
    check: {
        width: '100%',
        minWidth: 120,
        marginTop: theme.spacing(1.5),
        marginBottom: theme.spacing(1),

    },
    form: {
        width: '100%',
        marginTop: theme.spacing(0),
        minWidth: 120,
    },
    submit: {
        margin: theme.spacing(3, 0, 2),
        maxWidth: 200,
        backgroundColor: Color.primary('blue'),
        color: Color.primary('white'),
        '&:hover': {
            color: Color.primary('blue'),
            backgroundColor: Color.primary('white'),
        },
    },
    deleteButton: {
        margin: theme.spacing(3, 0, 2),
        maxWidth: 200,
        color: Color.primary('eliteRed'),
        backgroundColor: Color.primary('white'),
        '&:hover': {
            backgroundColor: Color.primary('eliteRed'),
            color: Color.primary('white'),
        },
    },
    dateTimeSelector: {
        marginTop: theme.spacing(0),
        width: '100%',
        minWidth: 150,
    },
    inputBoxInColumn: {
        marginBottom: theme.spacing(1),
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
    },
    dateInput: {
        margin: theme.spacing(0),
    },
    text: {
        textAlign: 'center',
    },
    submitButtonrow: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-around',
    },
    textStyle: {
        fontWeight: 'bold',
    },
    warningMessage: {
        padding: theme.spacing(0),
        color: Color.primary('eliteRed'),
    }
}));