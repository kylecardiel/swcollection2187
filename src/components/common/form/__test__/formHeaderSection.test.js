import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { FormHeaderSection } from 'components/common/form/formHeaderSection';

describe('FormHeaderSection', () => {

    const text = 'This is test text';
    const FormHeaderSectionComp = <FormHeaderSection text={text} textColor={'white'}/>;

    afterEach(cleanup);

    it('renders component', () => {
        const { asFragment } = render(FormHeaderSectionComp);
        expect(asFragment()).toMatchSnapshot();
    });

    describe('renders className', () => {
        it('adds correct className', () => {
            const { getByTestId } = render(FormHeaderSectionComp);
            expect(getByTestId('FormHeaderId')).toHaveClass('MuiGrid-root makeStyles-headerContainer-109 MuiGrid-container MuiGrid-spacing-xs-1');
        });
    });
});
