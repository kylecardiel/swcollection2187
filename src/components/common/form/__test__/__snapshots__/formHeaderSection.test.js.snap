// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHeaderSection renders component 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-headerContainer-1 MuiGrid-container MuiGrid-spacing-xs-1"
    data-testid="FormHeaderId"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="makeStyles-textSizeChange-105 makeStyles-textSizeChange-107"
        data-testid="headerTextId"
      >
        This is test text
      </div>
    </div>
  </div>
</DocumentFragment>
`;
